src/loader/BlendLoader.py

```diff
@@ -73,28 +106,88 @@ def run(self):
                                 "\nIf your ID exists, but not supported, please append a new pair of "
                                 "{type ID(folder name): parameter name} to the 'known_datablock_names' dict. Use this "
                                 "for finding your parameter name: " + str(dir(data_from)))
-        newly_loaded_objects = list(set(bpy.context.scene.objects) - previously_loaded_objects)
-        self._set_properties(newly_loaded_objects)
+
+    def _find_datablock_name_match_in_blendfile(self, blend_file_data, data_block_name):
+        """
+        Finds the corresponding datablock name in loaded .blend file.
+        .blend file uses slightly different string name for Datablocks, includes
+        underscores and extra s/'es'. For example /Mesh datablock name is matched to 
+        meshes attribute of .blend file. GreasePencil is matched to grease_pencils and so on.
+
+        :param blend_file_data: contents of loaded .blend file
+        :param data_block_name: Datablock name of a .blend file
+        """
+        blend_file_datablock_names = dir(blend_file_data)
+        index = -1
+        for i, attr in enumerate(blend_file_datablock_names):
+            # remove underscores like grease_pencils -> greasepencils and match with GreasePencil
+            # datablock
+            attr = attr.replace('_', '')
+            if data_block_name.lower() in attr:
+                index = i
+                break
+        if index == -1:
+
+            # The Datablock is valid but the .blend file does not contain the datablock. Likely
+            # version not supported.
+            raise Exception("Could not match Datablock {} in the .blend file. please Verify that \
+                            the .blend file version supports {} ID. Current \
+                             Blender API Version {}".format(data_block_name, data_block_name, bpy.app.version_string))
+
+        return blend_file_datablock_names[index]
+
+    def _set_datablock_properties(self, resource):
+        """
+        Sets the custom properties of **non object** resources like materials,
+        textures, images, etc.
+
+        Note: Some datablock types like bpy.types.Light, bpy.types.Mesh, bpy.types.Camera etc
+        are wrapped in bby.types.Object which act as a container of these object. In that case
+        setting the properties of the container object does not set the properties of underlying datablock like
+        camera and vice versa. Setting the bpy.data.objects["Light"] and bpy.data.lights["light"] can
+        have different properties. This function sets properties of all types materials, lights,
+        cameras even if they are loaded as an object. For Objects that wraps Meshes use  self._set_properties
+        instead
+        """
+        properties = self.config.get_raw_dict("add_properties", {})
+        for key, value in properties.items():
+            resource[key] = value           
```

Comment:
I think it does not take into account that setting custom properties is done via key starting with `cp_` but this prefix is not assigned as a part of the key, so here [key=key[:3]](https://github.com/DLR-RM/BlenderProc/blob/master/src/loader/LoaderInterface.py#L46) should be also done.